datafusion/common/src/scalar/mod.rs

```diff
@@ -2710,27 +2774,6 @@ impl From<String> for ScalarValue {
     }
 }
 
-// TODO: Remove this after changing to Scalar<T>
```

Comment:
These were added in https://github.com/apache/arrow-datafusion/pull/7893 and not yet released so this is not a breaking API change